### PR TITLE
prevent wrong type inference on doctrine dbal 2.x

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/staabm/phpstan-dba/issues/94#issuecomment-1016534736

should prevent 

> * `\Doctrine\DBAL\Connection::query` returns `\Doctrine\DBAL\Driver\Statement` and not `Doctrine\DBAL\Result`. I get errors like this: `Call to an undefined method Doctrine\DBAL\Result<array<int|string, int>>::fetchColumn().`
> we use version 2.13

uses https://getcomposer.org/doc/07-runtime.md#installed-versions

### CODE REVIEW

**Comment by staabm** [comment]
@mitelg could you check whether this PR fixes the errors you see regarding `Call to an undefined method Doctrine\DBAL\Result` ?

File: src/Extensions/DoctrineConnectionDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 namespace staabm\PHPStanDba\Extensions;
 
+use Composer\InstalledVersions;
+use Composer\Semver\VersionParser;
```
Comment (Seldaek) [inline src/Extensions/DoctrineConnectionDynamicReturnTypeExtension.php:8]: You need to require composer/semver if you want to use `satisfies`.

File: composer.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
 	"require": {
 		"php": "^8.0",
 		"ext-mysqli": "*",
+		"composer-runtime-api": "^2.2",
```
Comment (staabm) [inline composer.json:8]: 2.0 should be enough

File: src/Extensions/DoctrineConnectionDynamicReturnTypeExtension.php (hunk 1 above)
Comment (staabm) [reply src/Extensions/DoctrineConnectionDynamicReturnTypeExtension.php:8]: Thx for the hint. Interesting that the build is green even though the dependency is not declared

Comment (Seldaek) [reply src/Extensions/DoctrineConnectionDynamicReturnTypeExtension.php:8]: Probably something else you depend on requires it, so it works but not super failsafe :)

**Comment by mitelg** [comment]
code wise this looks good, but didn't had the chance yet, to check it

**Pull request merged**